### ISSUE TITLE
Signature help showing when it shouldn't

### ISSUE THREAD

**Issue by faldor20** [ocaml/ocaml-lsp#1162]
Signature help seems to show up event when it clearly shouldn't
Example:
```ocaml
let add a b =a + b in !!
!!
add 1 1;; !!
!!
```
Placing the cursor at any of the locations marked `!!` in this list and pressing "space" will cause the signature help popup to appear.

Signature help should not appear when:
- The scope has closed with a `;;` or `in`  
- The parameters of the function are already filled: eg: `add 1 2` should not trigger signature help after `2` because both of the params are satisfied

**Comment by roycrippen4** [comment]
This is still an issue and @faldor20's repro works exactly as stated.

<details>
    <summary>Example 1</summary>

![Image](https://github.com/user-attachments/assets/fc17226e-d685-4e95-9fad-d0232d04bb7b)


</details>

<details>
    <summary>Example 2</summary>

![Image](https://github.com/user-attachments/assets/a7ee50bc-ab1b-44c7-ae69-c8ab2efa2234)



</details>

Is there a way to disable signature help entirely with lsp configuration settings?

**Comment by voodoos** [comment]
Indeed, signature help still needs more fine tuning. There is no custom ocaml-lsp-server to turn it off, but editors might provide that as a standard setting.
